Shadow Sorcerer - Hound of Ill Omen Companion

BraabHimself opened the pull request:
Automatically add the Hound of Ill Omen companion sheet when a Shadow sorcerer gains level 6, just like the Hexblade's Accursed Specter.

Remove the Hound companion sheet when the level is 5 or less.

Also, correct a minor typo

safety-orange commented:
Thank you for the submission. I agree that it would be good to have the code add the hound automatically as a companion. I do have two notes about this though:

1. You are changing the text in the top left of the companion page to read "Monstrosity" (i.e. where it normally says "Companion" or "Familiar"). I think you might have confused it for the creature type (e.g. Beast, Monstrosity, Undead), which will already be set correctly when the creature is added to the sheet. The "Comp.Type" field is set to "Companion" by default but can be changed to whatever the player feels fits best. It is pretty limited in how much can fit in that field. Perhaps "Summoned" or just "Creature"? Probably also best not to make that field read-only.

2. You are setting the Temp HP field to automatically have a value **which can't be changed**. This is probably a bad idea as people will want to reduce the temporary hit points as the creature takes damage, which is impossible in how you set it up. Although people can keep track of the temp HP loss in the Current Hit Points / Wounds fields. Thinking about it a bit more, it will be hard to fully automated this for people. Maybe better to set it every time the sorcerer level changes, thus with a `changeeval` function.

BraabHimself commented:
I copied the way that the Hexblade's Accursed Specter works, so these problems are present there, too.

1. I've changed the Comp.Type field to be summoned and _not_ read only.
2. I've removed the use of the event.value calculation for the companion's temporary HP field. Instead, I simply set the value of the field. Then, whenever the sorcerer's level changes, the companion's temporary HP field is adjusted. This makes it so the field is editable _and_ scales the the player's sorcerer level.

If you agree with these changes, I'll make similar changes to the Hexblade's Accursed Specter.
